scrapers/freshmaker.py

```diff
@@ -0,0 +1,84 @@
+        """
+        Main function that runs the Freshmaker scraper
+        :param since: a string representing a datetime to start scraping data from
+        :return: None
+        """
+        if since:
+            log.info('Start date: {0} does not apply to Freshmaker scraper!'.format(since))
+        log.info('Starting initial load of Freshmaker events')
+        # Initialize session and base url
+        session = retry_session()
+        fm_url = self.freshmaker_url
+        self.query_api_and_update_neo4j(session, fm_url)
+        log.info('Initial load of Freshmaker events complete!')
+
+    def query_api_and_update_neo4j(self, session, fm_url):
+        """
+        Scrapes Freshmaker API and uploads data to Neo4j
+        :param start_date: a string representing a datetime to start scraping data from
+        :return: None
+        """
+
+        while True:
+            log.debug('Querying {0}'.format(fm_url))
+            rv_json = session.get(fm_url, verify=False).json()
+            for fm_event in rv_json['items']:
+                event = FreshmakerEvent.create_or_update(dict(
+                    id_=fm_event['id'],
+                    event_type_id=fm_event['event_type_id'],
+                    message_id=fm_event['message_id'],
+                    search_key=fm_event['search_key'],
+                    state=fm_event['state'],
+                    state_name=fm_event['state_name'],
+                    state_reason=fm_event['state_reason'],
+                    url=fm_event['url']
+                ))[0]
+
+                for build in fm_event['builds']:
+                    # To handle a faulty container build in Freshmaker
+                    if not build['build_id']:
+                        continue
+
+                    cb_params = dict(
+                        id_=build['id'],
+                        build_id=build['build_id'],
+                        dep_on=build['dep_on'],
+                        event_id=build['event_id'],
+                        name=build['name'],
+                        original_nvr=build['original_nvr'],
+                        rebuilt_nvr=build['rebuilt_nvr'],
+                        state=build['state'],
+                        state_name=build['state_name'],
+                        state_reason=build['state_reason'],
+                        time_submitted=timestamp_to_datetime(build['time_submitted']),
+                        type_=build['type'],
+                        type_name=build['type_name'],
+                        url=build['url']
+                    )
+                    if build['time_completed']:
+                        cb_params['time_completed'] = timestamp_to_datetime(
+                            build['time_completed'])
+                    cb = ContainerBuilds.create_or_update(cb_params)[0]
+                    cb.triggered_by_freshmaker_event.connect(event)
+                    event.triggers_container_builds.connect(cb)
+
```

Comment:
We'll need to create the relationships with the Errata advisories too. We can do this in this PR after we merge #25 or we can do it in a separate PR.